main.rb

```diff
@@ -0,0 +1,80 @@
+require 'terminal-table/import'
+require_relative 'constant'
+require_relative 'validator'
+require_relative 'grocery_item'
+
+
+# prompt: Add items to the cart
+puts "Please enter all the items purchased separated by a comma"
+
+inputs = gets.chomp
+
+# Converting items string to an array
+items = inputs.split(',')
+
+# remove unwanted spaces in the items
+items.map! { |item| item.strip }
+
+# valid-input validation
+unless Validation::validateInput(items)
+    exit
+end
+
+# mapping over items and converting the array of strings to symbols
+items.map! { |item| Constants::GROCERY_ITEMS_MAPPINGS[item.to_sym] }
+
+# get discounted price and total bill
+def get_discounted_price(groceryItemsColl)
+    return 0 if groceryItemsColl.empty?
+
+    item = groceryItemsColl[0]
+    count = groceryItemsColl.size
+
+    if (item.sale_price.nil?)
+        return count * item.unit_price
+    end
+
+    quotient = count / item.sale_qty
+    remainder = count % item.sale_qty
+
+    return (quotient * item.sale_price) + (remainder * item.unit_price)
+end
+
+def get_discounted_bill(groceryItems)
+    groupHash = groceryItems.group_by { |item| item.name }
+    result = groupHash.reduce(0) { |acc, (name, list)|
+        acc + get_discounted_price(list)
+    }
+    return result
+end
+
+def pricing_table(items)
+
+    return [] if items.empty?
+
+    result = []
+    items_group = items.group_by{ |item| item.name }
+    items_group.each { |name, list|
+    result << [ name, list.size, "$#{get_discounted_price(list)}"]    
+    }
+    result
+end
+
+total_bill = items.reduce(0) { |acc, item| 
```

Comment:
```
get_discounted_price
get_discounted_bill
pricing_table
```
These methods exist in global space without any class, what we can do here is we can have a class that could do this for us.
Something like a `class Bill`
Also, try to make use of encapsulation & expose only the necessary methods.